src/audio/copier/copier.c

```diff
@@ -1408,6 +1413,23 @@ static void copier_dma_cb(void *arg, enum notify_id type, void *data)
 	/* callback for one shot copy */
 	if (cd->hd->copy_type == COMP_COPY_ONE_SHOT)
 		host_one_shot_cb(cd->hd, bytes);
+
+	/* apply attenuation */
+	if (cd->attenuation) {
+		if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
+			sink = buffer_acquire(cd->hd->local_buffer);
+		else
+			sink = buffer_acquire(cd->hd->dma_buffer);
+
+		frames = bytes / get_sample_bytes(sink->stream.frame_fmt);
+		frames = frames / sink->stream.channels;
+
+		ret = apply_attenuation(dev, cd, sink, frames);
+		if (ret < 0)
+			comp_cl_dbg(dev, "copier_dma_cb() apply attenuation failed! %d", ret);
+
+		buffer_release(sink);
```

Comment:
you'd also need a buffer_writeback() no?

Reply:
yes, added with:
buffer_stream_writeback(sink, bytes);